kinto/core/views/heartbeat.py

```diff
@@ -11,10 +14,35 @@ def get_heartbeat(request):
     """Return information about server health."""
     status = {}
 
+    def heartbeat_check(name, func):
+        status[name] = False
+        status[name] = func(request)
+
+    # Start executing heartbeats concurrently.
     heartbeats = request.registry.heartbeats
-    for name, callable in heartbeats.items():
-        status[name] = callable(request)
+    pool = ThreadPoolExecutor(max_workers=max(1, len(heartbeats.keys())))
+    futures = []
+    for name, func in heartbeats.items():
+        future = pool.submit(heartbeat_check, name, func)
+        future.__heartbeat_name = name  # For logging purposes.
+        futures.append(future)
+
+    # Wait for the results, with timeout.
+    seconds = float(request.registry.settings['heartbeat_timeout_seconds'])
+    done, not_done = wait(futures, timeout=seconds)
+
+    # A heartbeat is supposed to return True or False, and never raise.
+    # Just in case, go though results to spot any potential exception.
+    for future in done:
```

Comment:
ok I see, so there's the convention that the heartbeat functions should catch all errors. But we never know what might happen in external heartbeat functions.

Is that a behaviour we want to keep ? 

e.g. do we want the global heartbeat to completely fail when one heartbeat is producing an error, or do we want to lg that error and flag that backend to false in the result ?